src/components/Launcher.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { Component } from 'react'
+import { translate } from '../lib/I18n'
+import Autosuggest from 'react-autosuggest';
+import { rawIntent } from '../lib/intents'
+
+const INTENT_VERB = 'OPEN'
+const INTENT_DOCTYPE = 'io.cozy.suggestions'
+
+class Launcher extends Component {
+  constructor (props) {
+    super(props)
+    
+    this.state = {
+      query: '',
+      suggestions: []
+    }
+    
+    this.sourcesNode = null
+    this.sources = []
+  }
+  
+  componentDidMount () {
+    // The launcher has one or more sources that provide suggestions. These sources are iframes into other apps, provied by thee intent system.
+    // Since we need to call the sources whenever the query changes, we are taking manual control over the intent process.
+    window.addEventListener('message', this.onMessageFromSource)
+    
+    rawIntent(INTENT_VERB, INTENT_DOCTYPE)
+    .then(intent => {
+      const { services } = intent.attributes
+      if (!services) return
+      
+      services.forEach(service => {
+        // we're injecting the iframes, but they will stay invisible
+        const iframe = document.createElement('iframe')
```

Comment:
I don't like messing with the DOM too much in components: would you mind to instantiate these iframes using JSX in the render method?

Reply:
I agree with @goldoraf, shouldn't we put this code part into `/lib/intents.js` ? The intent lib in cozy-client actually deals with all the DOM interactions.

Reply:
👍 I'll go with @goldoraf 's suggestion and put it in the render method. I really don't want to legitimize this intent workaround, or encourage anyone else to use it.